### PR TITLE
Feature credential detail

### PR DESCRIPTION
**Pull request by wadeking98**
added credential verification page to orgbook, as well as some minor UI tweaks for the entity result page.

### CODE REVIEW

File: src/components/search/SearchBar.vue [hunk 1]
```diff
@@ -100,6 +100,9 @@ export default class SearchBar extends Vue {
   }
 
   async autocomplete(q: string): Promise<void> {
+    if (q.length < 2) {
```
Comment (amanji) [inline src/components/search/SearchBar.vue:103]: Might want to also add a hint that says to type more than 2 characters, but I can add that to an issue

File: src/router/index.ts [hunk 2]
```diff
@@ -10,12 +10,19 @@ const routes: Array<RouteConfig> = [
...
   },
+
   {
     path: "/entity/:sourceId",
     name: "Entity",
     component: () =>
       import(/* webpackChunkName: "entity" */ "@/views/Entity.vue"),
   },
+  {
+    path: "/credential/:credentialID",
```
Comment (amanji) [inline src/router/index.ts:21]: Should probably be a child path on `/entity/:sourceId/`

File: src/store/modules/credential-detail.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    id: string
+  ): Promise<void> {
+    try {
+      const res = await credentialService.getFormattedCredential(id);
+      commit("setSelectedCredential", res.data);
+    } catch (e) {
+      console.error(e);
+    }
+  },
+  async fetchPresID(
```
Comment (amanji) [inline src/store/modules/credential-detail.ts:45]: Nitpicking but use camelCase so `ID` should be `Id`. There are similar instances of naming across this file that we may want to update as well

File: src/store/modules/credential-type.ts [hunk 4]
```diff
@@ -55,12 +56,26 @@ const actions = {
       commit("setTypes", []);
     }
   },
+  async fetchCredentialTypeByID(
```
Comment (amanji) [inline src/store/modules/credential-type.ts:59]: Is this required of we already get a list of credential types?

File: src/utils/date-filter.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import moment from "moment";
```
Comment (amanji) [inline src/utils/date-filter.ts:1]: Duplicate date filter as one above

File: src/views/CredentialView.vue [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <v-container :fluid="$vuetify.breakpoint.smAndDown" class="pa-8">
+    <CredentialDetail />
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import CredentialDetail from "@/components/entity/credentialDetail/CredentialDetail.vue";
```
Comment (amanji) [inline src/views/CredentialView.vue:9]: Should this be in the Credentials components?

File: src/interfaces/api/v4/credential.interface.ts [hunk 7]
```diff
@@ -43,3 +46,36 @@ export interface ICredentialDisplayType {
...
+  inactive: boolean;
+  latest: boolean;
+  revoked: boolean;
+  revoked_date?: Date;
+  credential_id: string;
+  credential_type: ICredentialType;
+  addresses: ICredential[];
+  names: ICredentialName;
+  attributes: ICredentialAttribute[];
+  local_name: ICredentialName;
```
Comment (amanji) [inline src/interfaces/api/v4/credential.interface.ts:62]: I'm not sure if we want to carry over the concept of local name vs remote name from the previous implementation of OrgBook

File: src/interfaces/api/v2/topic.interface.ts [hunk 8]
```diff
@@ -31,7 +31,16 @@ export interface ITopic extends IApiResource {
   related_from: number[];
 }
 
+export interface ITopicLocalName {
```
Comment (amanji) [inline src/interfaces/api/v2/topic.interface.ts:34]: See my other comment about local vs remote name. It often causes a lot of confusion and is only used in particular circumstances. We can circle back on the discussion for this.

**Review by amanji — CHANGES_REQUESTED** [review]

File: src/interfaces/api/v4/credential.interface.ts (hunk 7 above)
Comment (wadeking98) [reply src/interfaces/api/v4/credential.interface.ts]: okay I removed local/remote name from credential and topic interfaces

File: src/components/entity/CredentialItem.vue [hunk 9]
```diff
@@ -22,20 +22,20 @@
...
         <div v-if="reason">
           <span>Reason:&nbsp;</span>
           <span>{{ reason }}</span>
         </div>
-        <div>
+        <div v-if="!expired">
           <v-icon small>{{ mdiShieldCheckOutline }}</v-icon>
-          <span class="fake-link vertical-align-middle"
-            >Credential verified</span
+          <router-link :to="`${baseRoute}/credential/${credId}`"
```
Comment (amanji) [inline src/components/entity/CredentialItem.vue:31]: Can also pass an object with the route name and the url parameters

File: src/components/entity/credentialDetail/CredentialDetail.vue [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+    this.setLoading(true);
+
+    const { sourceId, credentialId } = this.$route.params;
+    this.sourceId = sourceId
+    if (sourceId && credentialId) {
+      await Promise.all([
+        this.fetchSelectedCredential(credentialId),
+        this.fetchPresId(credentialId),
+      ]);
+      console.log(this.getPresentationId);
```
Comment (amanji) [inline src/components/entity/credentialDetail/CredentialDetail.vue:278]: Remove console statements

**Review by amanji — CHANGES_REQUESTED** [review]

File: src/components/entity/CredentialItem.vue (hunk 9 above)
Comment (wadeking98) [reply src/components/entity/CredentialItem.vue]: change completed

File: src/components/entity/credentialDetail/CredentialDetail.vue (hunk 10 above)
Comment (wadeking98) [reply src/components/entity/credentialDetail/CredentialDetail.vue]: removed

**Review by amanji — APPROVED** [review]

**Pull request merged**